### DISCUSSION TITLE
Da W a KWh - utilizzo Integrale

### DISCUSSION THREAD

**Discussion by Matty1184** [indomus/forum#3226] [HUB: Home Assistant]
Ciao a tutti, vorrei monitorare i KWh utilizzati nella giornata dalla mia pompa di calore, ho un sensore energia che mi mostra i Watt istantanei e ho creato un Integral che mi converta i W in kWh
ho utilizzato 
metodo: trapezoidal
precisione 2
max sub-interval 1min
a parte il valore che mi "converte" il sensore creato è incrementale, ovvero è partito da zero quando l'ho creato e continua a salire.
ma anche il valore che va ad incrementare non è corretto, esempio:
W istantanei 29 per almeno mezza giornata, il sensore mi incrementa in 3 ore 1kwh
W istantai dai 10 ai 3000 per mezz'ora oscillando in quel valore, il sensore ha incrementato in quell'ora oltre 3Kwh ma se il consumo max istantaneo per mezz'ora è 3000w al max avrei dovuto leggere indicativamente 1,5Kwh.
sto ragionando in modo errato?
cosa mi consigliate di utilizzare?
lo scopo di questo dato è poi mandarlo in un contatore a fasce orarie e calcolarmi  l'ipotetico costo giornaliero e/o mensile ma questo avverrà dopo aver risolto questo

**Comment by trebestie** [comment]
Vado un po' a memoria perchè è roba che ho configurato tempo fa
step 1 crei un aiutante con integrale a somma sinistra di Riemann[*] che legga i dati del sensore che ti interessa
step 2 crei uno o più misuratori di utilità che leggono i dati del sensore creato prima e si resettano sulla base delle tue esigenze
Per esempio io ho giorno, settimana, mese, anno e alltime

Spero di avere capito la tua richiesta

Buona serata

[*] [mi sembra la modalità più corretta per questo tipo di consumi](https://www.home-assistant.io/integrations/integration/#left)

**Comment by indomus** [comment]
> Ciao a tutti, vorrei monitorare i KWh utilizzati nella giornata dalla mia pompa di calore, ho un sensore energia che mi mostra i Watt istantanei e ho creato un Integral che mi converta i W in kWh ho utilizzato metodo: trapezoidal precisione 2 max sub-interval 1min a parte il valore che mi "converte" il sensore creato è incrementale, ovvero è partito da zero quando l'ho creato e continua a salire. ma anche il valore che va ad incrementare non è corretto, esempio: W istantanei 29 per almeno mezza giornata, il sensore mi incrementa in 3 ore 1kwh W istantai dai 10 ai 3000 per mezz'ora oscillando in quel valore, il sensore ha incrementato in quell'ora oltre 3Kwh ma se il consumo max istantaneo per mezz'ora è 3000w al max avrei dovuto leggere indicativamente 1,5Kwh. sto ragionando in modo errato? cosa mi consigliate di utilizzare? lo scopo di questo dato è poi mandarlo in un contatore a fasce orarie e calcolarmi l'ipotetico costo giornaliero e/o mensile ma questo avverrà dopo aver risolto questo

https://indomus.it/guide/calcolare-il-consumo-di-energia-in-kwh-a-partire-dalla-potenza-istantanea-in-watt-con-home-assistant/

**Comment by Matty1184** [comment]
Grazie a entrambi, 
ho variato ora l'integrale utilizzando "Sinistra" ma non sto capendo una cosa...
il valore restituito è sempre incrementale?

**Reply by KC-inDomus** [reply]
assolutamente sì

**Reply by Matty1184** [reply]
OK, chiarissimo...
quindi per il dato giornaliero che strada posso prendere?

**Reply by Scialla** [reply]
utility meter, prendi spunto da qui:
https://indomus.it/progetti/configurare-dei-contabilizzatori-di-consumo-energetico-su-home-assistant-v2/
